File: scripts/merge_bedpe.R
#!/usr/bin/env R

# Usage 
# File1 column File2 column File3 column ... FileX column out_file
# Column is 1-indexed

args     <- commandArgs(trailingOnly=TRUE)

out_file <- args[length(args)]
args     <- args[-length(args)]

files  <- args[seq(1,length(args),2)]
column <- args[seq(1,length(args),2)+1]
column <- as.numeric(as.character(column))

dfs <- lapply(seq(files),function(x){
  df <- read.table(file = files[x])
  colnames(df)[column[x]] <- 'merge_by'
  df
})

df_final <- Reduce(function(df1, df2) merge(df1, df2, by = "merge_by"),dfs)
df_final <- df_final[order(as.numeric(gsub('loop_','',df_final$merge_by))),]
df_final <- df_final[,-1]

write.table(x = df_final,file = out_file,quote = FALSE,sep="\t",row.names = FALSE,col.names = FALSE)

File: scripts/merge_bedpe.py
#!/usr/bin/env python3

#import pandas as pd
import sys
import pandas as pd

def convert_to_integer(l):
  try:
    return([int(x) for x in l])
  except ValueError:
    return(False)

# Parse arguments

def main():
  # Error return:
  err_msg = '*** Usage is merge_bedpe.py file1 column file2 column file3 column ... ***\n'
  
  to_merge = sys.argv[1:]
  to_merge =  [(to_merge[x],to_merge[x+1]) for x in range(0,len(to_merge),2)]
  
  if len(to_merge) %2 != 0:
    sys.stderr.write(err_msg)
    sys.exit(1)

  
  columns = convert_to_integer([x[1] for x in to_merge])
  if not columns:
    sys.stderr.write("Column numbers can't be converted to integers\n" + err_msg)
    sys.exit(1)
    
  files   = [x[0] for x in to_merge]
  
  
  dfs = []
  for x,f in enumerate(files):
    df                     = pd.read_table(f, header=None)
    colnames               = [x for x in df.columns]
    colnames[columns[x]-1] = 'merge_column'
    df.columns             = colnames
    dfs.append(df)






main()